src/swarm/neo/client/mixins/RequestCore.d

```diff
@@ -259,11 +259,42 @@ public template RequestCore ( RequestType request_type_, ubyte request_code,
 
         /***********************************************************************
 
-            Object for acquiring resources required by the request.
+            Object for acquiring shared resources required by the request.
 
         ***********************************************************************/
 
-        public SerializableReferenceType!(Object) request_resources;
+        public Object shared_resources;
+
+        /***********************************************************************
+
+            Note: this struct is essentially deprecated. It is only needed in
+            order to maintain the public API of Context. In the next major
+            release, this wrapper struct can be removed.
+
+        ***********************************************************************/
+
+        private struct RequestResources
+        {
+            private Object* request_resources;
```

Comment:
Why is this a pointer?

Reply:
So that the return value of `get` always reflects the value of `shared_resources` in the outer context.

Reply:
It's a bit theoretical, but you could do:
```D
auto rr = context.request_resources();
do_something(rr.get());
context.shared_resources = something_else;
do_something(rr.get()); // should reflect the changed value
```

Reply:
I see. Fair enough.